### ISSUE TITLE
QKSMS Freezes when deleting large conversation (10,000+ Messages)

### ISSUE THREAD

**Issue by frogoscar** [moezbhatti/qksms#640]
### DESCRIPTION

QKSMS becomes ANR when deleting large conversation (10,000+ Messages).
### STEPS
1. Have a conversation of 10,000+ messages
2. Delete this conversation
3. During the deletion of this large conversation (taking more than 20 minutes), try to close and re-open QKSMS
### EXPECTED

QKSMS can be re-opened
### OBSERVATIONS

QKSMS can not be re-opened, ANR message is showing
### Concerned source code

```
public static void startDelete(ConversationQueryHandler handler, int token, boolean deleteAll, Collection<Long> threadIds) {
        synchronized (sDeletingThreadsLock) {
            if (DELETEDEBUG) {
                Log.v(TAG, "Conversation startDelete sDeletingThreads: " + sDeletingThreads);
            }
            if (sDeletingThreads) {
                Log.e(TAG, "startDeleteAll already in the middle of a delete", new Exception());
            }
            //MmsApp.getApplication().getPduLoaderManager().clear();
            sDeletingThreads = true;

            for (long threadId : threadIds) {
                Uri uri = ContentUris.withAppendedId(Threads.CONTENT_URI, threadId);
                String selection = deleteAll ? null : "locked=0";

                handler.setDeleteToken(token);
                handler.startDelete(token, new Long(threadId), uri, selection, null);

                DraftCache.getInstance().setDraftState(threadId, false);
            }
        }
    }
```

in data/Conversation.java

**Comment by frogoscar** [comment]
While the standard Messages App of Android do not have ANR. It has a ProgressDialog during the deletion of large conversation.

**Comment by moezbhatti** [comment]
This should be fixed in QKSMS 3.0
